pkg/controller/decomission.go

```diff
@@ -30,6 +30,25 @@ import (
 // checkForPoolDecommission validates the spec of the tenant and it's status to detect a pool being removed
 func (c *Controller) checkForPoolDecommission(ctx context.Context, key string, tenant *miniov2.Tenant, tenantConfiguration map[string][]byte) (*miniov2.Tenant, error) {
 	var err error
+	// duplicate status.pools first
+	haveRepeatStatusPool := false
+	statusPoolsDuplicateMap := map[string]struct{}{}
```

Comment:
Actually the map is used for removing duplicates, so may be it should be called `distinctStatusPoolsMap`